### PR TITLE
Add simple recon_trace:quick/2 for beginners

### PR DESCRIPTION
**Pull request by dch**
It's a big leap for a newcomer to tracing to go straight to `recon_trace:calls/3` and friends. This should allow even newcomers to Erlang to use tracing from even the earliest days.

I can revert the whitespace pruning if that's important to you.

The change produces no more dialyzer whining than the current code.

### CODE REVIEW

**Comment by ferd** [comment]
I have to consider it. The problem with this is that it puts more load on the system for no good reason. Tracing *all* calls (even local ones) without any filtering whatsoever can be heavy enough to put a node in geopardy before the rate-limiter can kick in. Think of tracing `gen_server` on a very busy node, for example.

**Comment by dch** [comment]
sure, this is always important. possibly setting a maximum trace value would be one option. I use something similar all the time in local debug/devel cycle. I guess what we'd want is to support tracing only user modules, and not OTP ones, for example. I think having a smaller step, in some form, is a Good Thing. At the end of the day, tracing is risky (all things in moderation!) and people will need to acquire an understanding of that either through personal experience, or thinking carefully.

**Comment by ferd** [comment]
I've thought about it a bit, and I'm not in favor of including this change under its current form for the following reasons:

- it is a big leap for newcomers to jump into tracing, but it *should be so* (to some extent) because bad tracing can adversely impact a node either in performance or memory.
- this of itself isn't a major problem, but an 'easy' solution to tracing intended for beginners should be *safe* by default because the newcomer may not be trusted to assess risk properly. 
- Having a default of all pids is okay -- it's the default right now
- Having a default of all functions at a local scope is *not okay*. This generate a very large amount of logs when any specifically busy module (say `gen.erl`) gets traced, and on busy systems it could be enough to topple a system over.

In short, the two things this call adds (tracing all functions of a module, and doing so at a local scope), are things I consciously didn't allow to be done by default for safety reasons when someone works on a production system, which recon is intended for as its top priority.

I am not willing to revert this choice and merge this PR in, as I feel the risks would be higher than someone reading the documentation and trying things out from a safe point first. This is a place where I see safety trumping ease of use.

A thing that could be done is support a format of `{Mod, Fun}`, which would be equivalent to `{Mod, Fun, '_'}` which already represents a better level of isolation on how things work, but I'm not sure it's that much of an improvement to begin with.

**Comment by dch** [comment]
thanks @ferd for the clarification. I'll keep this as a .erlangrc then
